### PR TITLE
Add module template generation

### PR DESCRIPTION
**Pull request by clebs**
**Description**

Changes proposed in this pull request:

- A template is generated when a module is pushed to a registry.
- Added small validation improvements.

**Related issue(s)**
#1316

### CODE REVIEW

File: cmd/kyma/alpha/create/module/module.go [hunk 1]
```diff
@@ -24,9 +26,9 @@ func NewCmd(o *Options) *cobra.Command {
 	}
 
 	cmd := &cobra.Command{
-		Use:   "module MODULE_NAME MODULE_VERSION <CONTENT_PATH> [flags]",
-		Short: "Creates a module bundled as an OCI image from the contents of the given path",
-		Long: `Use this command to create a Kyma module.
+		Use:   "module OCI_IMAGE_NAME MODULE_VERSION <CONTENT_PATH> [flags]",
+		Short: "Creates a module bundled as an OCI image with the given from the contents of the given path",
```
Comment (NHingerl) [inline cmd/kyma/alpha/create/module/module.go:30]: "with the given from the contents" sounds weird

Comment (clebs) [reply cmd/kyma/alpha/create/module/module.go:30]: Yes, I think I forgot a word in there...

Comment (clebs) [reply cmd/kyma/alpha/create/module/module.go]: Fixed on previous PR to avoid more confusion on early adopters.

**Review by jakobmoellersap — APPROVED** [review]

File: docs/gen-docs/kyma_alpha_create_module.md [hunk 2]
```diff
@@ -26,10 +26,12 @@ kyma alpha create module OCI_IMAGE_NAME MODULE_VERSION <CONTENT_PATH> [flags]
 ## Flags
 
 ```bash
+      --channel string         Channel to use for the module template. (default "stable")
       --clean                  Remove the mod-path folder and all its contents at the end.
   -c, --credentials string     Basic authentication credentials for the given registry in the format user:password
       --insecure               Use an insecure connection to access the registry.
       --mod-path string        Specifies the path where the component descriptor and module packaging will be stored. If the path already has a descriptor use the overwrite flag to overwrite it (default "./mod")
+  -o, --output string          File to output the module template if the module is uploaded to a registry (default "template.yaml")
```
Comment (majakurcius) [inline docs/gen-docs/kyma_alpha_create_module.md:34]:
```suggestion
  -o, --output string          File to which to output the module template if the module is uploaded to a registry (default "template.yaml")
```
or `File to output the module template to if the module...`

**Review by jakobmoellersap — APPROVED** [review]

**Review by majakurcius — DISMISSED** [review]

**Review by jakobmoellersap — DISMISSED** [review]

**Review by majakurcius — APPROVED** [review]

**Review by jakobmoellersap — APPROVED** [review]

**Pull request merged**